src/Provider/AbstractProvider.php

```diff
@@ -135,17 +176,46 @@ public function setScopes(array $scopes)
         $this->scopes = $scopes;
     }
 
+    /**
+     * Get a new random string to use for auth state.
+     *
+     * @param  integer $length
+     * @return string
+     */
+    protected function getRandomState($length = 32)
+    {
+        $generator = $this
+            ->getRandomFactory()
+            ->getMediumStrengthGenerator();
+
+        return $generator->generateString($length);
+    }
+
     public function getAuthorizationUrl(array $options = [])
     {
-        $this->state = isset($options['state']) ? $options['state'] : md5(uniqid(rand(), true));
+        if (empty($options['state'])) {
```

Comment:
I thought `empty()` with throw a notice error if an array key didn't exists and `isset()` wouldn't... but I just check it and it doesn't. Doh! Lol.

But the other reason I was going to propose is that the lib uses `isset` originally but doing a search in the code, the lib uses `isset` and `empty` quite interchangeably.